### PR TITLE
remove the debugging statements

### PR DESCRIPTION
**Pull request by mwnicks**
### What

Removed the debugging statements from the nginx debugging

### How to review

Read the changes.

### Who can review

Anyone but me

### CODE REVIEW

**Review by redhug1 — CHANGES_REQUESTED** [review]
The changes look sensible and tests pass.
However there are some issues elsewhere in the codebase.
Any chance you can sort the following out:
Config.go, line 49 - Variable 'json' collides with imported package name
users_test.go, line 840 - Use camel case instead of snake case
users.go, lines 18, 119, 188, 263 - Unhandled error
tokens.go, line 16 - Unhandled error
service/healthcheck/healthcheck.go, lines 23, 34, 45, 51 - Unhandled error

Thanks.

File: api/tokens.go [hunk 1]
```diff
@@ -13,7 +13,11 @@ import (
 
 //TokensHandler uses submitted email address and password to sign a user in against Cognito and returns a http handler interface
 func (api *API) TokensHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
-	defer req.Body.Close()
+	defer func() {
+		if err := req.Body.Close(); err != nil {
+			err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
```
Comment (redhug1) [inline api/tokens.go]: err is not being logged, maybe replace this line with:
log.Event(ctx, models.BodyClosedFailedDescription, log.Error(err), log.ERROR)

File: api/users.go [hunk 2]
```diff
@@ -15,7 +15,11 @@ import (
 
 //CreateUserHandler creates a new user and returns a http handler interface
 func (api *API) CreateUserHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
-	defer req.Body.Close()
+	defer func() {
+		if err := req.Body.Close(); err != nil {
+			err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
```
Comment (redhug1) [inline api/users.go]: err is not being logged, maybe replace this line with:
log.Event(ctx, models.BodyClosedFailedDescription, log.Error(err), log.ERROR)

File: api/users.go [hunk 3]
```diff
@@ -116,7 +120,11 @@ func (api *API) GetUserHandler(ctx context.Context, w http.ResponseWriter, req *
 
 //UpdateUserHandler updates a users details in Cognito and returns a http handler interface
 func (api *API) UpdateUserHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
-	defer req.Body.Close()
+	defer func() {
+		if err := req.Body.Close(); err != nil {
+			err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
```
Comment (redhug1) [inline api/users.go]: err is not being logged, maybe replace this line with:
log.Event(ctx, models.BodyClosedFailedDescription, log.Error(err), log.ERROR)

File: api/users.go [hunk 4]
```diff
@@ -185,7 +193,11 @@ func processUpdateCognitoError(ctx context.Context, err error, errContext string
 
 //ChangePasswordHandler processes changes to the users password
 func (api *API) ChangePasswordHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
-	defer req.Body.Close()
+	defer func() {
+		if err := req.Body.Close(); err != nil {
+			err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
```
Comment (redhug1) [inline api/users.go]: err is not being logged, maybe replace this line with:
log.Event(ctx, models.BodyClosedFailedDescription, log.Error(err), log.ERROR)

File: api/users.go [hunk 5]
```diff
@@ -260,7 +272,11 @@ func (api *API) ChangePasswordHandler(ctx context.Context, w http.ResponseWriter
 
 //PasswordResetHandler requests a password reset email be sent to the user and returns a http handler interface
 func (api *API) PasswordResetHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
-	defer req.Body.Close()
+	defer func() {
+		if err := req.Body.Close(); err != nil {
+			err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
```
Comment (redhug1) [inline api/users.go]: err is not being logged, maybe replace this line with:
log.Event(ctx, models.BodyClosedFailedDescription, log.Error(err), log.ERROR)

**Review by redhug1 — CHANGES_REQUESTED** [review]
a few fixes needed ...

**Review by redhug1 — CHANGES_REQUESTED** [review]
some fixes needed ...

File: api/tokens.go (hunk 1 above)
Comment (mwnicks) [reply api/tokens.go]: The models.NewError function logs the error. This is too make sure all errors are logged when they have been generated in to the required struct.
[https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23](https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23)

File: api/users.go (hunk 2 above)
Comment (mwnicks) [reply api/users.go]: The models.NewError function logs the error. This is too make sure all errors are logged when they have been generated in to the required struct.
[https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23](https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23)

File: api/users.go (hunk 3 above)
Comment (mwnicks) [reply api/users.go]: The models.NewError function logs the error. This is too make sure all errors are logged when they have been generated in to the required struct.
[https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23](https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23)

File: api/users.go (hunk 4 above)
Comment (mwnicks) [reply api/users.go]: The models.NewError function logs the error. This is too make sure all errors are logged when they have been generated in to the required struct.
[https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23](https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23)

File: api/users.go (hunk 5 above)
Comment (mwnicks) [reply api/users.go]: The models.NewError function logs the error. This is too make sure all errors are logged when they have been generated in to the required struct.
[https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23](https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23)

File: api/tokens.go (hunk 1 above)
Comment (redhug1) [reply api/tokens.go]: Ah thanks.
Then in which case, the line with:
err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
should really be:
_ = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)

to clearly show that code is not interested in return value from NewError ?

File: api/users.go (hunk 2 above)
Comment (redhug1) [reply api/users.go]: Ah thanks.
Then in which case, the line with:
err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
should really be:
_ = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)

to clearly show that code is not interested in return value from NewError ?

File: api/users.go (hunk 3 above)
Comment (redhug1) [reply api/users.go]: Ah thanks.
Then in which case, the line with:
err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
should really be:
_ = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)

to clearly show that code is not interested in return value from NewError ?

File: api/users.go (hunk 4 above)
Comment (redhug1) [reply api/users.go]: Ah thanks.
Then in which case, the line with:
err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
should really be:
_ = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)

to clearly show that code is not interested in return value from NewError ?

File: api/users.go (hunk 5 above)
Comment (redhug1) [reply api/users.go]: Ah thanks.
Then in which case, the line with:
err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
should really be:
_ = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)

to clearly show that code is not interested in return value from NewError ?

File: api/tokens.go (hunk 1 above)
Comment (mwnicks) [reply api/tokens.go]: Yeah thats a good point.
I've updated now.

**Review by redhug1 — APPROVED** [review]

**Pull request merged**